#!/usr/bin/env python
PACKAGE = "uchile_speech_pocketsphinx"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("vad_postspeech",    int_t,    0, "Num of silence frames to keep after from speech to silence", 50,  0, 100)
gen.add("vad_prespeech", int_t, 0, "Num of speech frames to keep before silence to speech",    20, 0,100)
gen.add("vad_startspeech",    int_t,    0, "Num of speech frames to trigger vad from silence to speech.",  10,0,100)
gen.add("vad_threshold",   double_t,   0, "Threshold for decision between noise and silence frames. Log-ratio between signal level and noise level.",  2.0,0.0,10.0)


exit(gen.generate(PACKAGE, "uchile_speech_pocketsphinx", "SpeechRecognition"))